internal/fs/watcher.go

```diff
@@ -0,0 +1,280 @@
+package fs
```

Comment:
Can this file be called file_watcher. go ? Also can the name of package name be expanded from fs to something more descriptive ?

Reply:
Do you think that `filesystem/file_watcher.go` would be a more accurate name?

Reply:
> Do you think that `filesystem/file_watcher.go` would be a more accurate name?

Yes, that would be most excellent !